### ISSUE TITLE
Issue when using WCF service in mvc

### ISSUE THREAD

**Issue by NST77** [MarimerLLC/cslaforum#232]
I have a WCF project with CSLA and I have published this service and MVC application in my IIS Server, but it's throwing one error when calling the service from Web app

> System.ServiceModel.Security.SecurityNegotiationException: The caller was not authenticated by the service. ---> System.ServiceModel.FaultException: The request for security token could not be satisfied because authentication failed.

This is my service web.config

`<system.serviceModel>
    <serviceHostingEnvironment multipleSiteBindingsEnabled="true" />
    <services>
      <service name="Csla.Server.Hosts.WcfPortal" behaviorConfiguration="returnFaults">
        <endpoint binding="wsHttpBinding" bindingConfiguration="wsHttpBinding_IWcfPortal" contract="Csla.Server.Hosts.IWcfPortal" />
      </service>
      <service name="Csla.Server.Hosts.Mobile.WcfPortal" behaviorConfiguration="returnFaults">
        <endpoint binding="basicHttpBinding" bindingConfiguration="basicHttpBinding_IWcfPortal" contract="Csla.Server.Hosts.Mobile.IWcfPortal" />
      </service>
    </services>
    <bindings>
      <basicHttpBinding>
        <binding name="basicHttpBinding_IWcfPortal" maxReceivedMessageSize="2147483647" maxBufferPoolSize="2147483647" maxBufferSize="2147483647">
          <readerQuotas maxBytesPerRead="2147483647" maxArrayLength="2147483647" maxStringContentLength="2147483647" maxNameTableCharCount="2147483647" maxDepth="2147483647" />
        </binding>
      </basicHttpBinding>
      <wsHttpBinding>
        <binding name="wsHttpBinding_IWcfPortal" maxReceivedMessageSize="2147483647">
          <readerQuotas maxBytesPerRead="2147483647" maxArrayLength="2147483647" maxStringContentLength="2147483647" maxNameTableCharCount="2147483647" maxDepth="2147483647" />
        </binding>
      </wsHttpBinding>
    </bindings>
    <behaviors>
      <serviceBehaviors>
        <behavior name="returnFaults">
          <serviceDebug includeExceptionDetailInFaults="true" />
          <serviceMetadata httpGetEnabled="true" />
        </behavior>
      </serviceBehaviors>
    </behaviors>
  </system.serviceModel>`

 I have consume this service in my MVC web.config like this.

`<add key="CslaAuthentication" value="Csla" />
  <add key="CslaDataPortalProxy" value="Csla.DataPortalClient.WcfProxy, Csla" />
  <add key="CslaDataPortalUrl" value="http://www.someurl.net/WcfFullPortal.svc" />`

Then after I have called service directly from my service project, that was working fine
`<add key="CslaDataPortalUrl" value="http://localhost:11170/WcfFullPortal.svc" />`

And also tried with published IP address, that also working fine
`<add key="CslaDataPortalUrl" value="http://10.4.56.75/WcfFullPortal.svc" />`

Only problem when I use published host name(http://www.someurl.net/WcfFullPortal.svc). And I have open this URL in by browser that was working.

**Comment by rockfordlhotka** [comment]
The web site and service are two ASP.NET sites running on the same IIS server, and the web site can call the service using localhost, but not the external domain name. Is that correct?

I'm not enough of an IT Pro to know what's going on there, but I have often encountered issues when trying to do this sort of thing - domain configurations, routing tables, DNS entries - all seem to get talked about, people (IT Pros) change things and look irritated, and then things start working.

Not useful I know, but that's my typical experience.

**Comment by NST77** [comment]
HI, Thanks for your valuable replay.
I have fixed the issue. It's a fault from my side 

I have changed `Csla.DataPortalClient.WcfProxy`  to `Csla.DataPortalClient.HttpProxy`
and ` <add key="CslaDataPortalUrl" value="http://www.someurl.net/WcfFullPortal.svc" />` to ` <add key="CslaDataPortalUrl" value="http://www.someurl.net/DataPortal/PostAsync" />`

And also added `Csla.ApplicationContext.User = new Csla.Security.UnauthenticatedPrincipal();` in my Global.asax.cs

And one another doubt, like this it's only taking Async calls?. In my project some palaces I have calling any method on the BO that is not Async it's returning 

> isSync == true
error

And my client app is an MVC Web site.
